core/evm/sputnik/sputnik.go

```diff
@@ -1,53 +1,38 @@
 package sputnik
 
 import (
-	"bytes"
 	"errors"
-
-	"github.com/gallactic/gallactic/core/account"
-	"github.com/gallactic/gallactic/core/blockchain"
-	"github.com/gallactic/gallactic/core/state"
-	"github.com/gallactic/gallactic/crypto"
-	"github.com/gallactic/gallactic/txs/tx"
-
+	"fmt"
 	"math/big"
 
 	"github.com/ethereumproject/go-ethereum/common"
-	"github.com/gallactic/gallactic/common/binary"
 	"github.com/gallactic/sputnikvm-ffi/go/sputnikvm"
 )
 
-//Execute for executing virtual machine
-func Execute(bc *blockchain.Blockchain, cache *state.Cache, caller, callee *account.Account, tx *tx.CallTx, gas *uint64, isDeploying bool) ([]uint8, error) {
-
+func Execute(adapter Adapter) (Output, error) {
+	var out Output
 	var retError error
 
-	var addrCaller common.Address
-
-	var addrCallee common.Address
-
-	callerBytes := caller.Address().RawBytes()[2:22]
-	addrCaller.SetBytes(callerBytes)
-
-	calleeBytes := callee.Address().RawBytes()[2:22]
-	addrCallee.SetBytes(calleeBytes)
+	if adapter.calleeAddress() == nil && len(adapter.GetData()) == 0 {
+		return out, errors.New("Zero Bytes of Codes")
+	}
 
 	transaction := sputnikvm.Transaction{
-		Caller:   addrCaller,
+		Caller:   adapter.callerAddress(),
 		GasPrice: new(big.Int).SetUint64(0),
-		GasLimit: new(big.Int).SetUint64(tx.GasLimit()),
-		Address:  &addrCallee,
-		Value:    new(big.Int).SetUint64(tx.Amount()),
-		Input:    tx.Data(),
-		Nonce:    new(big.Int).SetUint64(tx.Caller().Sequence),
+		GasLimit: new(big.Int).SetUint64(adapter.GetGasLimit()),
+		Address:  adapter.calleeAddress(),
+		Value:    new(big.Int).SetUint64(adapter.GetAmount()),
+		Input:    adapter.GetData(),
+		Nonce:    new(big.Int).SetUint64(adapter.GetNonce()),
 	}
 
 	header := sputnikvm.HeaderParams{
 		Beneficiary: *new(common.Address),
-		Timestamp:   uint64(bc.LastBlockTime().Unix()),
-		Number:      new(big.Int).SetUint64(bc.LastBlockHeight()),
+		Timestamp:   adapter.TimeStamp(),
+		Number:      new(big.Int).SetUint64(adapter.LastBlockNumber()),
 		Difficulty:  new(big.Int).SetUint64(0),
-		GasLimit:    new(big.Int).SetUint64(tx.GasLimit()),
+		GasLimit:    new(big.Int).SetUint64(adapter.GetGasLimit()),
 	}
 
 	vm := sputnikvm.NewFrontier(&transaction, &header)
```

Comment:
* We should use `Gallactic` Patch instead of `Frontier`
* The sputnik should initialize when the node starts not on each transaction to improve performance.  Look #61